types/process.luau

```diff
@@ -5,34 +5,127 @@ export type SpawnOptionsStdioKind = "default" | "inherit" | "forward" | "none"
+
+	* `cwd` - The current working directory for the process
+	* `env` - Extra environment variables to give to the process
+	* `shell` - Whether to run in a shell or not - set to `true` to run using the default shell, or a string to run using a specific shell
+	* `stdio` - How to treat output and error streams from the child process - see `SpawnOptionsStdioKind` and `ExecuteOptionsStdio` for more info
+	* `stdin` - Optional standard input to pass to executed child process
+]=]
+export type ExecuteOptions = SpawnOptions & {
 	stdio: (SpawnOptionsStdioKind | SpawnOptionsStdio)?,
 	stdin: string?, -- TODO: Remove this since it is now available in stdio above, breaking change
 }
 
+--[=[
+	@class ChildProcessReader
+	@within Process
+
+	A reader class to read data from a child process' streams in realtime.
+]=]
+local ChildProcessReader = {}
+
+--[=[
+	@within ChildProcessReader
+
+	Reads a chunk of data (specified length or a default of 8 bytes at a time) from 
+	the reader as a string. Returns nil if there is no more data to read.
+
+	This function may yield until there is new data to read from reader, if all data
+	till present has already been read, and the process has not exited.
+
+	@return The string containing the data read from the reader
+]=]
+function ChildProcessReader:read(chunkSize: number?): string?
+	return nil :: any
+end
+
+--[=[
+	@within ChildProcessReader
+
+	Reads all the data currently present in the reader as a string.
+	This function will yield until the process exits.
+
+	@return The string containing the data read from the reader
+]=]
+function ChildProcessReader:readToEnd(): string
+	return nil :: any
+end
+
+--[=[
+	@class ChildProcessWriter
+	@within Process
+
+	A writer class to write data to a child process' streams in realtime.
+]=]
+local ChildProcessWriter = {}
+
+--[=[
+	@within ChildProcessWriter
+
+	Writes a buffer or string of data to the writer.
+
+	@param data The data to write to the writer
+]=]
+function ChildProcessWriter:write(data: buffer | string): ()
+	return nil :: any
+end
+
 --[=[
 	@interface SpawnResult
 	@within Process
 
-	Result type for child processes in `process.spawn`.
+	Result type for child processes in `process.create`.
+
+	This is a dictionary containing the following values:
+
+	* `stdin` - A writer to write to the child process' stdin - see `ChildProcessWriter` for more info
+	* `stdout` - A reader to read from the child process' stdout - see `ChildProcessReader` for more info
+	* `stderr` - A reader to read from the child process' stderr - see `ChildProcessReader` for more info
+	* `kill` - A function that kills the child process
+	* `status` - A function that yields and returns the exit status of the child process
+]=]
+export type ChildProcess = {
+	stdin: typeof(ChildProcessWriter),
+	stdout: typeof(ChildProcessReader),
+	stderr: typeof(ChildProcessReader),
+	kill: () -> ();
+	status: () -> { ok: boolean, code: number }
+}
```

Comment:
How come `ChildProcess` is not a class but `ChildProcessWriter` and `ChildProcessReader` are?

Reply:
`ChildProcess` does not include any methods, and is only a type which describes the shape of the object returned by `process.create`. 

`ChildProcessWriter` and `ChildProcessReader` are classes which contain methods on them. Furthermore, they _have_ to be classes, since it is not possible to attach moonwave doc comments to methods within types.